src/main/java/com/google/sps/servlets/AuthServlet.java

```diff
@@ -11,20 +11,31 @@
 import com.google.appengine.api.users.UserService;
 import com.google.appengine.api.users.UserServiceFactory;
 import com.google.gson.Gson;
+import com.google.inject.Inject;
 import com.google.sps.data.AuthData;
 
 @WebServlet("/auth")
 public class AuthServlet extends HttpServlet {
+  UserService userService;
+
+  @Inject
+  AuthServlet(UserService userService) {
+    this.userService = userService;
+  }
+
+  public AuthServlet() {
+    userService = UserServiceFactory.getUserService();
+  }
```

Comment:
It seems that you don't use dependency injection properly. You need to specify what gets injected like here: https://github.com/marcbury-g/step/blob/master/appengine-di-skeleton/src/main/java/com/example/step/MyServletModule.java

So in your case you would write a module with a method 
```
@Provides
Userservice provideUserService() { return UserServiceFactory.getUserService(); }
```
Then you don't need this default constructor.